dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/test/InMemoryByteSinkFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.test;
+
+import com.google.edwmigration.dumper.application.dumper.io.OutputHandle;
+import com.google.edwmigration.dumper.application.dumper.io.OutputHandleFactory;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+public class InMemoryByteSinkFactory implements OutputHandleFactory {
+
+  private final Map<String, DummyOutputHandle> producedHandlers = new HashMap<>();
```

Comment:
Related to previous comment; it's somewhat confusing that at first read, this factory of in-memory byte sinks doesn't actually seem to factorize any such sinks, until one goes and reads `DummyOutputHandle` to understand that it is just a wrapper around an in-memory byte sink.

Reply:
You're right. I renamed this class to `InMemoryOutputHandleFactory` to make it clear. It's debatable whether `InMemoryOutputHandle` class should at all be visible outside of the factory, as we might be able to perform the test assertions on concrete handles via the factory object itself, but I left it as is since it gives more flexibility and is a much more straight design.